eslint-config-azuretools/index.js

```diff
@@ -0,0 +1,40 @@
+module.exports = {
+    "env": {
+        "es6": true,
+        "node": true
+    },
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "project": "tsconfig.json",
+        "sourceType": "module"
+    },
+    "plugins": [
+        "@typescript-eslint",
+        "import"
+    ],
+    "extends": [
+        "eslint:recommended",
+        "plugin:@typescript-eslint/recommended",
+        "plugin:@typescript-eslint/recommended-requiring-type-checking"
+    ],
+    "rules": {
+        "no-unused-vars": "off",
+        "no-useless-escape": "off",
+        "no-inner-declarations": "off",
+        "no-case-declarations": "off",
+        "@typescript-eslint/prefer-regexp-exec": "off",
+        "@typescript-eslint/no-unused-vars": [
+            "error",
+            { "argsIgnorePattern": "^_" }
+        ],
+        "@typescript-eslint/no-inferrable-types": "off",
+        "@typescript-eslint/unbound-method": "off",
+        "@typescript-eslint/no-unnecessary-type-assertion": "off",
+        "@typescript-eslint/restrict-template-expressions": "off",
+        "@typescript-eslint/no-namespace": "off",
+        "@typescript-eslint/ban-types": "off",
+        "@typescript-eslint/require-await": "off",
+        "no-constant-condition": ["error", { "checkLoops": false }],
+        "eqeqeq": ["error", "always"]
```

Comment:
I used functions' ESLint config as the base for this. The only change is this line which was requested [here](https://github.com/microsoft/vscode-cosmosdb/pull/1872#discussion_r661784591)

Reply:
Two things:
1. We have test specific ESLint configs for some (I know at least functions and SWA) extensions. Could we create a shared config for these as well?
2. I recently added two rules to the SWA extension in this PR: microsoft/vscode-azurestaticwebapps/pull/456. As Eric mentioned on that PR, It would be great to include these rules in the shared config. Do you want to include these rules in the shared config now, or later once the config is implemented and used by the extensions?
 
Note: adding this rule requires tweaking [(see here for what I added for SWA)](https://github.com/microsoft/vscode-azurestaticwebapps/blob/main/test/.eslintrc.js#L6) to the test ESLint config to prevent errors in tests. I quickly tested out adding these rules to functions and the required tweak was similar.

Reply:
> We have test specific ESLint configs for some (I know at least functions and SWA) extensions. Could we create a shared config for these as well?

Our ESLint configs for tests are less standardized than those for production code. I'm open to coming up with a single config for these but I think it might be overkill. I'm curious to see what the rest of the team thinks about this though.

For reference, here's some of our test config files to compare:
https://github.com/microsoft/vscode-azureappservice/blob/main/test/.eslintrc.js
https://github.com/microsoft/vscode-azurefunctions/blob/main/test/.eslintrc.js
https://github.com/microsoft/vscode-cosmosdb/blob/main/test/.eslintrc.js
https://github.com/microsoft/vscode-azurestaticwebapps/blob/main/test/.eslintrc.js
https://github.com/microsoft/vscode-azurevirtualmachines/blob/main/test/.eslintrc.js

> I recently added two rules to the SWA extension in this PR: Add no-restricted-imports rule to help prevent wrong ext imports vscode-azurestaticwebapps#456. As Eric mentioned on that PR, It would be great to include these rules in the shared config. Do you want to include these rules in the shared config now, or later once the config is implemented and used by the extensions?

Thanks for bringing this up, I'll go ahead and add them now